### DISCUSSION TITLE
Dependency Injection Bindings in AXAML

### DISCUSSION THREAD

**Discussion by michalss** [AvaloniaUI/Avalonia#15432] [General]
Hi all,

Is there a way to directly use DI service with bindings like this please ? Compailer complain about it and it wont build. WPF can do it why avalonia cant ? Please help.. thx a lot

```axaml
    <StackPanel>
        <Button Content="Add Log" Command="{Binding AddLogCommand }" />
        <ListBox ItemsSource="{Binding LoggingService.LogItems}"></ListBox>
    </StackPanel>
```

```csharp
   public ILoggingService LoggingService;
    public MainWindowViewModel(ILoggingService loggingService)
    {
        LoggingService = loggingService;
    }
```

`  MainWindow.axaml(20, 18): [AVLN:0004] Unable to resolve property or method of name 'LoggingService' on type 'SharingModelView.ViewModels.MainWindowViewModel'. Line 20, position 18.`

**Comment by BAndysc** [comment]
LoggingService needs to be a property, not a field

**Reply by michalss** [reply]
Cau you please be more specific?

**Reply by workgroupengineering** [reply]
https://github.com/AvaloniaUI/Avalonia.Samples/pull/64/files
https://github.com/AvaloniaUI/Avalonia/issues/15313

**Reply by thevortexcloud** [reply]
For more general info on DI there was also this guide recently published.

https://docs.avaloniaui.net/docs/guides/implementation-guides/how-to-implement-dependency-injection

But yes, you can bind to any property, including ones with an injected value.

**Reply by michalss** [reply]
Thank you but it does not really answer the question :)

**Reply by rabbitism** [reply]
> Thank you but it does not really answer the question :)

To answer your question, it has to be 

public ILoggingService LoggingService { get; }

**Reply by michalss** [reply]
this is something i was looking for it could seems to be a simple thing but i just could not figure it out :)
